v7/static_comments/static_comments.py

```diff
@@ -0,0 +1,246 @@
+    author_url = None
+    author_ip = None  # should not be published by default
+    date_utc = None  # should be set via set_utc_date()
+    date_localized = None  # should be set via set_utc_date()
+
+    # set by _process_comments():
+    indent_levels = None  # use for formatting comments as tree
+    indent_change_before = 0  # use for formatting comments as tree
+    indent_change_after = 0  # use for formatting comments as tree
+
+    # The meaning of indent_levels, indent_change_before and
+    # indent_change_after are the same as the values in utils.TreeNode.
+
+    def __init__(self, site, owner, id, parent_id=None):
+        """Initialize comment.
+
+        site: Nikola site object;
+        owner: post which 'owns' this comment;
+        id: ID of comment;
+        parent_id: ID of comment's parent, or None if it has none.
+        """
+        self._owner = owner
+        self._config = site.config
+        self.id = id
+        self.parent_id = parent_id
+
+    def set_utc_date(self, date_utc):
+        """Set the date (in UTC). Automatically updates the localized date."""
+        self.date_utc = utils.to_datetime(date_utc)
+        self.date_localized = utils.to_datetime(date_utc, self._config['__tzinfo__'])
+
+    def formatted_date(self, date_format):
+        """Return the formatted localized date."""
+        return utils.LocaleBorg().formatted_date(date_format, self.date_localized)
+
+    def hash_values(self):
+        """Return tuple of values whose hash to consider for computing the hash of this comment."""
+        return (self.id, self.parent_id, self.content, self.author, self.author_url, self.date_utc)
+
+    def __repr__(self):
+        """Returns string representation for comment."""
+        return '<Comment: {0} for {1}; indent: {2}>'.format(self.id, self._owner, self.indent_levels)
+
+
+class StaticComments(SignalHandler):
+    """Add static comments to posts."""
+
+    # Used to parse comment headers
+    _header_regex = re.compile('^\.\. (.*?): (.*)')
+
+    def _compile_content(self, compiler_name, content, filename):
+        """Compile comment content with """
+        if compiler_name not in self.site.compilers:
+            _LOGGER.error("Cannot find page compiler '{0}' for comment {1}!".format(compiler_name, filename))
+            exit(1)
+        compiler = self.site.compilers[compiler_name]
+        if compiler_name == 'rest':
+            content, error_level, _ = compiler.compile_string(content)
+            if error_level >= 3:
+                _LOGGER.error("Restructured text page compiler ({0}) failed to compile comment {1}!".format(compiler_name, filename))
+                exit(1)
+            return content
+        elif compiler_name == 'ipynb':
+            return compiler.compile_string(content)
+        else:
+            try:
+                return compiler.compile_to_string(content)  # this is a non-standard function! must not be available with any page compiler!
```

Comment:
The WordPress compiler provides `compile_to_string`, while `rest` and `ipynb` provide `compile_string` (though `rest` doesn't simply return a string, but a tuple).